pkg/build/api/types.go

```diff
@@ -231,6 +231,11 @@ type BuildOutput struct {
 	// a Docker image repository to push to. Failure to find the To will result in a build error.
 	To *kapi.ObjectReference `json:"to,omitempty"`
 
+	// pushSecret is the name of a Secret that would be used for setting
+	// up the authentication for executing the Docker push to authentication
+	// enabled Docker Registry (or Docker Hub).
+	PushSecret string `json:"pushSecret,omitempty"`
```

Comment:
Needs to be consistent with Paul's latest, which is "SecretName".  You should not merge this until that discussion settles upstream.